tools/generate_release_notes.py

```diff
@@ -0,0 +1,215 @@
+    if new_user.login not in users:
+        if new_user.name is None:
+            users[new_user.login] = new_user.login
+        else:
+            users[new_user.login] = new_user.name
+
+for commit in tqdm(all_commits, desc='Getting commiters and authors'):
+    # committer can be None?
+    if commit.committer:
+        add_to_users(users, commit.committer)
+        committers.add(users[commit.committer.login])
+
+    # Users that deleted their accounts will appear as None
+    # So annoying
+    if commit.author is None:
+        warn('Could not find author of commit :' + commit.sha)
+        continue
+
+    add_to_users(users, commit.author)
+    authors.add(users[commit.author.login])
+    if None in authors:
+        import pdb; pdb.set_trace()
+        print('Why do I get here')
+# this gets found as a commiter
+committers.discard('GitHub Web Flow')
+authors.discard('Azure Pipelines Bot')
+assert None not in authors
+assert None not in committers
+
+highlights = OrderedDict()
+
+highlights['New Feature'] = {}
+highlights['Improvement'] = {}
+highlights['Bugfix'] = {}
+highlights['API Change'] = {}
+highlights['Deprecations'] = {}
+highlights['Build Tool'] = {}
+other_pull_requests = {}
+
+for pull in tqdm(g.search_issues(f'repo:{GH_USER}/{GH_REPO} '
+                                 f'merged:>{previous_tag_date.isoformat()} '
+                                 'sort:created-asc'),
+                 desc='Iterating through for Pull Requests'):
+    pr = repository.get_pull(pull.number)
+    if pr.merge_commit_sha in all_commit_sha:
+        summary = pull.title
+        for r in pr.get_reviews():
+            if r.user.login not in users:
+                users[r.user.login] = r.user.name
+            reviewers.add(users[r.user.login])
+        for key, key_dict in highlights.items():
+            pr_title_prefix = (key + ': ').lower()
+            if summary.lower().startswith(pr_title_prefix):
+                key_dict[pull.number] = {
+                    'summary': summary[len(pr_title_prefix):]}
+                break
+        else:
+            other_pull_requests[pull.number] = {
+                'summary': summary
+            }
+
+
+# add Other PRs to the ordered dict to make doc generation easier.
+highlights['Other Pull Request'] = other_pull_requests
+
+
+
+def name_sorting_key(name):
+    name = [v for v in name.split() if v[0] in string.ascii_letters]
+    if len(name) > 0:
+        return name[-1]
+
+
+# Now generate the release notes
+announcement_title = "Announcement: scikit-image {}".format(version)
+print(announcement_title)
+print("="*len(announcement_title))
+
+print("""
+We're happy to announce the release of scikit-image v{version}!
+
+scikit-image is an image processing toolbox for SciPy that includes algorithms
+for segmentation, geometric transformations, color space manipulation,
+analysis, filtering, morphology, feature detection, and moreself.
+""".format(version=version))
+
+if version.startswith('0.14'):
```

Comment:
Remove 0.14 specific code

Reply:
why? shouldn't this stay at least till the end of 2019?